### PR TITLE
Feature: PDF version(s) of the manual.

### PR DESCRIPTION
**Pull request by DaveMcEwan**
Use [pandoc](https://pandoc.org/MANUAL.html) in the mdgen and release workflows to generate development and release PDFs from `MANUAL.md`.
The main differences between development vs release versions are in the title page.
The mdgen workflow keeps the development version updated, sitting beside `MANUAL.md` in the repo root.
In this [example release](https://github.com/DaveMcEwan/svlint/releases/tag/v0.0.7-DONOTUSE), you can see a release version is packaged with the binaries for each platform, and a development version is in the source package/zip.

NOTE: Pandoc's Markdown is different from GitHub Flavored Markdown, so the number of changed files is rather large, sorry!
- The "unconfigured" regex for `re_forbidden_*` was treated as a footnote.
- Three dashes `---` indicating a horizontal rule were interpreted as YAML metadata opening delimiters. Replaced with space-separated asterisks with blank lines above and below.
- I've also used this PR to remove unnecessary indents from the "See also:" lists.
- Main changes are in the addition of the PDF, and the workflows.

Related changes:
- Added keyword "svls" which users might search for on crates.io.
- Added convenience rules to generate PDFs locally if pandoc is installed.
- Change format of LONG_VERSION (returned by `svlint --version` using release binaries).
  - Removed punctuation.
  - Space-separated fields are now usable with `cut`, for example `cut -d' ' -f3` gets the git hash.
  - Colon-separated sub-fields.
  - Build time complies with [ISO8601](https://en.wikipedia.org/wiki/ISO_8601) like `2023-02-20T20:09:53`.
- Used non-recursive Makefile variables.
  Just makes future developers try harder if they really want to break the Makefile.

@dalance If you accept this PR, would you make a new release?

### CODE REVIEW

**Comment by DaveMcEwan** [comment]
Sorry, I've just found the pandoc extension [`lists_without_preceding_blankline`](https://pandoc.org/MANUAL.html#extension-lists_without_preceding_blankline) so I'll try using that to reduce the diff.

**Comment by DaveMcEwan** [comment]
Apologies for the churn! Markdown and PDF now look a little better now, and the diff isn't too awful with the "Hide whitespace" box ticked.

**Comment by DaveMcEwan** [comment]
In a future PR, the last pushed commit must not change `MANUAL.md`, otherwise the add-and-commit step fails in a PR's checks.
That restriction is unchanged, but contributors are (before this PR) able to workaround it by running mdgen locally.
After this PR, if mdgen is ran locally but pandoc isn't, then the PDF can get out of date.
To keep the PDF up to date with `MANUAL.md`, either the contributor needs to run pandoc, or the last commit must be something that isn't altered by running mdgen.

I can see two options for dealing with this:
1. Accept it as a minor annoyance.
2. Remove pandoc PDF generation from the mdgen workflow.
  This doesn't affect the release workflow.

I'm tending towards the 2nd (removing from the mdgen workflow).
@dalance Do you have any advice?

**Comment by dalance** [comment]
How about moving `MANUAL.md` and `MANUAL.pdf` from master to another branch (ex. `manual` branch)?

**Comment by DaveMcEwan** [comment]
I'd be worried that it can be confusing which branch to look at...

Rationale for why a PDF is desired: When discussing rule/rulesets with colleagues, it's essential to know that we're both reading the same thing. A document which is frozen/released, time-stamped, and version-stamped makes that much easier.

Thinking about that rationale, it's not very important for development versions to have a PDF. As noted above, developers could still generate development PDFs with `make MANUAL-dev`. So, I've removed `MANUAL.pdf` from the PR and the mdgen workflow. Key points after the latest commit:
1. PDFs are only added to the ZIP releases. I've made a new [example release](https://github.com/DaveMcEwan/svlint/releases/tag/v0.0.8-DONOTUSE).
2. The mdgen workflow only checks that a PDF can still be created by converting from Markdown via pandoc. The release workflow uses the same version of pandoc.
3. Developers have no additional restrictions/annoyances wrt mdgen. As required currently, they must run mdgen locally and include the changes in their PR.
4. Developers can choose to create their own local PDF using either `make MANUAL-dev`, or `make MANUAL-release`.

TL;DR: I'm happy with this PR now, obviously pending any tips @dalance :p

**Comment by dalance** [comment]
PDF in release package seems to be reasonable.
I'll merge this PR.
